linkerd/app/inbound/src/http/tests.rs

```diff
@@ -37,6 +39,8 @@ where
 {
     let connect = svc::stack(connect)
         .push_map_target(|t: TcpEndpoint| Remote(ServerAddr(([127, 0, 0, 1], t.param()).into())))
+        .push_timeout(cfg.proxy.connect.timeout)
+        .push(svc::stack::BoxFuture::layer())
```

Comment:
I opted to add the timeout layer to the connect stack for all tests. If wanted, I could change `build_server` to take a connect stack instead of just a `Connect` struct so that the test is responsible for creating the connect stack used, and we don't have a timeout when we don't need it.

Reply:
hmm, rather than pushing _just_ the timeout stack, could we push the entire endpoint stack instead? i'm a little leery of having tests that test stacks that we don't actually build in a real proxy; i'd prefer to build the whole thing if it's possible.

@olix0r may disagree with me here though!

Reply:
also, if we're including timeouts in all the test stacks now we should _probably_ stick a `tokio::time::pause` in all the tests, so that they don't fail due to spurious timeouts? that probably won't ever happen, but if we don't want to exercise timeout behavior, we should probably turn them off i think...

Reply:
The reason I went with pushing just the timeout instead of the endpoint is right now `push_endpoint` is only a method for `Outbound` and we're dealing with `Inbound` here. By pushing just a timeout here we don't need to add a method to `Inbound` that would only be used by test code I think? Let me know if you had something else in mind.